src/jit/lowerxarch.cpp

```diff
@@ -2300,58 +2300,170 @@ void Lowering::ContainCheckSIMD(GenTreeSIMD* simdNode)
 //  Arguments:
 //     containingNode - The hardware intrinsic node which contains 'node'
 //     node - The node to check
+//     supportsRegOptional - On return, true if 'containingNode' supports regOptional operands; otherwise, false.
 //
 // Return Value:
 //    true if 'node' is a containable hardware intrinsic node; otherwise, false.
 //
-bool Lowering::IsContainableHWIntrinsicOp(GenTreeHWIntrinsic* containingNode, GenTree* node)
+bool Lowering::IsContainableHWIntrinsicOp(GenTreeHWIntrinsic* containingNode, GenTree* node, bool* supportsRegOptional)
 {
-    if (!node->OperIsHWIntrinsic())
+    NamedIntrinsic      containingIntrinsicID = containingNode->gtHWIntrinsicId;
+    HWIntrinsicCategory category              = Compiler::categoryOfHWIntrinsic(containingIntrinsicID);
+
+    // We shouldn't have called in here if containingNode doesn't support containment
+    assert((Compiler::flagsOfHWIntrinsic(containingIntrinsicID) & HW_Flag_NoContainment) == 0);
+
+    bool supportsAlignedSIMDLoads   = false;
```

Comment:
You should add a brief comment on each of these.

Reply:
Fixed.